### ISSUE TITLE
Invalid app name is allowed to deploy but then does not work when running sandbox in test env

### ISSUE THREAD

**Issue by defionscode** [architect/architect#762]
**Describe the issue**
Version: `Architect 6.3.3`

This is an interesting one. 
Given an 'improper' name like "myappV2"
You are allowed to:
1. Deploy via `arc deploy`
1. Run the sandbox via `arc sandbox`
But you cannot run sandbox in a test context which causes
```
× Error: [object Object]
```

I dug in and this is the unhandled err object
```js
  errors: [
    {
      keyword: 'pattern',
      dataPath: '.app[0]',
      schemaPath: '#/definitions/name/pattern',
      params: { pattern: '^[a-z][a-z|\\-|0-9]+$' },
      message: 'should match pattern "^[a-z][a-z|\\-|0-9]+$"'
    }
  ]
```
It's in `src/sandbox/read-arc.js` under the `@architect/sandbox` package

On a side note maybe we could add `debug(err)` in places where errors are handled to at least facilitate being able to get "raw" error objects?

**Comment by ryanblock** [comment]
Hmm, this should have been resolved in our subdependencies. I hate to ask you to unplug it and plug it back in, but can you see of the issue persists after rm -rf your node modules + lockfile and reinstalling?

**Comment by ryanblock** [comment]
Oh, and regarding that error, sorry, there was a bit of thrash as we rolled out our totally rewritten Architect parser a few weeks ago.

Parsing errors are now returned as strings, so we shouldn't need to be mucking about with meta error debugging. (See: https://github.com/architect/parser/blob/master/src/compat/validate.js#L36-L60)

**Comment by ryanblock** [comment]
Just checking back – this may have also been patched in 6.3.4 which was released Friday, any luck?
